image-sync/oc-mirror/deploy/image-sync-template.yml

```diff
@@ -0,0 +1,149 @@
+                env:
+                  - name: XDG_RUNTIME_DIR
+                    value: /etc/
+                volumeMounts:
+                - name: ocmirror-config
+                  mountPath: "/etc/ocmirror"
+                  readOnly: true
+                - name: pull-secrets-updated
+                  mountPath: "/etc/containers"
+                  readOnly: true
+                - name: workspace
+                  mountPath: "/workspace"
+              volumes:
+              - name: ocmirror-config
+                configMap:
+                  name: ocmirror-config
+              - name: workspace
+                emptyDir: {}
+              - name: pull-secrets-updated
+                emptyDir: {}
+              - name: pull-secret
+                csi:
+                  driver: secrets-store.csi.k8s.io
+                  readOnly: true
+                  volumeAttributes:
+                    secretProviderClass: "ocmirror-pullsecret"
+              serviceAccountName: image-sync
+    status: {}
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: ocmirror-config
+      namespace: ${NAMESPACE}
+    data:
+      imageset-config.yml: |
+        kind: ImageSetConfiguration
+        apiVersion: mirror.openshift.io/v1alpha2
+        storageConfig:
+          registry:
+            imageURL: ${TARGET_REGISTRY}.azurecr.io/mirror/oc-mirror-metadata
+            skipTLS: false
+        mirror:
+          platform:
+            channels:
+              - name: stable-4.16
+                minVersion: 4.16.0
+                maxVersion: 4.16.3
+                type: ocp
+            graph: true
+          additionalImages:
+          - name: registry.redhat.io/redhat/redhat-operator-index:v4.16
+          - name: registry.redhat.io/redhat/certified-operator-index:v4.16
+          - name: registry.redhat.io/redhat/community-operator-index:v4.16
+          - name: registry.redhat.io/redhat/redhat-marketplace-index:v4.16
+          - name: registry.redhat.io/redhat/redhat-operator-index:v4.15
+          - name: registry.redhat.io/redhat/certified-operator-index:v4.15
+          - name: registry.redhat.io/redhat/community-operator-index:v4.15
+          - name: registry.redhat.io/redhat/redhat-marketplace-index:v4.15
+          - name: registry.redhat.io/redhat/redhat-operator-index:v4.17
+          - name: registry.redhat.io/redhat/certified-operator-index:v4.17
+          - name: registry.redhat.io/redhat/community-operator-index:v4.17
+          - name: registry.redhat.io/redhat/redhat-marketplace-index:v4.17
+  - apiVersion: secrets-store.csi.x-k8s.io/v1
+    kind: SecretProviderClass
+    metadata:
+      name: ocmirror-pullsecret
+      namespace: ${NAMESPACE}
+    spec:
+      parameters:
+        clientID: "${OC_MIRROR_CLIENT_ID}"
+        cloudName: AzurePublicCloud
+        keyvaultName: ${KEYVAULT_NAME}
+        objects: |-
+          array:
+            - |
+              objectName: pull-secret
```

Comment:
consider providing `objectName` as a variable and set `objectAlias: pull-secret`

this way you can choose via `objectName` what secret to consume from KV but within the running pod it will be available in a "file" named `pull-secret`